src/main/java/ca/uhn/fhir/jpa/starter/CustomSearchNarrowingInterceptor.java

```diff
@@ -0,0 +1,38 @@
+package ca.uhn.fhir.jpa.starter;
+
+import com.auth0.jwt.JWT;
+import com.auth0.jwt.interfaces.DecodedJWT;
+import ca.uhn.fhir.interceptor.api.Interceptor;
+import ca.uhn.fhir.rest.api.server.RequestDetails;
+import ca.uhn.fhir.rest.server.interceptor.auth.AuthorizedList;
+import ca.uhn.fhir.rest.server.interceptor.auth.SearchNarrowingInterceptor;
+
+@Interceptor
+public class CustomSearchNarrowingInterceptor extends SearchNarrowingInterceptor {
+
+  private static final String OAUTH_CLAIM_NAME = System.getenv("OAUTH_CLAIM_NAME");
```

Comment:
I think we should hard-code the patient claim name. I don't see value is making this configurable and it would only lead to misconfiguration problems.

Reply:
I also think we may add other claims to the token at a later point.

Reply:
I will update with claim name "subject" in code